### PR TITLE
Implement events for Governance module

### LINKED ISSUE #8869: Implement events for Governance module
### Description

The module defines [these 4 events](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0074.md#events):
- Proposal Created
- Proposal Quorum Checked
- Proposal Outcome
- Proposal Voted

Proposal Creation Failed has been [removed in this PR](https://github.com/LiskHQ/lips/pull/460).

### Acceptance Criteria

- Implement all 4 events
- Any custom logic should be covered by unit tests

### PR DESCRIPTION
**Pull request by bobanm**
### What was the problem?

This PR resolves #8869 

### How was it solved?

Implemented 4 events:
- Proposal Created
- Proposal Outcome
- Proposal Quorum Checked
- Proposal Voted

Removed Proposal Creation Failed event, as it is [being removed from the LIP](https://github.com/LiskHQ/lips/pull/460)

### How was it tested?

No custom logic added to the events, so no tests.
The event functionality will be tested as part of the module unit tests.

### CODE REVIEW

File: framework/src/modules/governance/events/proposal_creation_failed.ts [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
- * no part of this software, including this file, may be copied, modified,
- * propagated, or distributed except according to the terms contained in the
- * LICENSE file.
- *
- * Removal or modification of this copyright notice is prohibited.
- */
-
-import { BaseEvent } from '../../base_event';
-
-export class ProposalCreationFailedEvent extends BaseEvent<void> {}
```
Comment (shuse2) [inline framework/src/modules/governance/events/proposal_creation_failed.ts:17]: why was this deleted?

File: framework/src/modules/governance/events/proposal_created.ts [hunk 2]
```diff
@@ -21,4 +21,31 @@ interface ProposalCreatedEventData {
...
+			fieldNumber: 3,
+		},
+	},
+};
+
+export class ProposalCreatedEvent extends BaseEvent<ProposalCreatedEventData> {
+	public schema = proposalCreatedEventDataSchema;
+
+	public log(ctx: EventQueuer, data: ProposalCreatedEventData): void {
+		this.add(ctx, data, [data.creator, Buffer.from(data.index.toString())]);
```
Comment (shuse2) [inline framework/src/modules/governance/events/proposal_created.ts:49]: data.index should be encoded with uint32 like
```
const index = Buffer.alloc(4);
index.writeUInt32BE(data.index);
```

File: framework/src/modules/governance/events/proposal_voted.ts [hunk 3]
```diff
@@ -22,4 +23,35 @@ interface ProposalVotedEventData {
...
+	type: 'object',
+	required: ['index', 'voterAddress', 'decision', 'amount'],
+	properties: {
+		index: {
+			dataType: 'uint32',
+			fieldNumber: 1,
+		},
+		voterAddress: {
+			dataType: 'bytes',
+			length: ADDRESS_LENGTH,
```
Comment (shuse2) [inline framework/src/modules/governance/events/proposal_voted.ts:37]: you can use `format: lisk32` here as well, and remove ADDRESS_LENGTH

File: framework/src/modules/governance/events/proposal_created.ts (hunk 2 above)
Comment (shuse2) [reply framework/src/modules/governance/events/proposal_created.ts:49]: applies to all other places as well

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/modules/governance/events/proposal_creation_failed.ts (hunk 1 above)
Comment (bobanm) [reply framework/src/modules/governance/events/proposal_creation_failed.ts:17]: As I wrote in the PR description, this event is being removed in the [upcoming PR](https://github.com/LiskHQ/lips/pull/460).

Based on my suggestion to move `hasEnded` check from `execute` to `verify` part of `CreateProposal` command, this event is no longer necessary.

**Review by shuse2 — APPROVED** [review]

**Review by mosmartin — APPROVED** [review]
